pennylane/beta/tapes/qnode.py

```diff
@@ -137,6 +144,25 @@ def __init__(self, func, device, interface="autograd", diff_method="best", **dif
         self.dtype = np.float64
         self.max_expansion = 2
 
+        self._caching = None
+        """float: number of device executions to store in a cache to speed up subsequent
+        executions. If set to zero, no caching occurs."""
+
+        if caching is not None:
+            warnings.warn(
+                "Caching mode activated. The quantum circuit being executed by the QNode must have "
+                "a fixed structure.",

```

Comment:
This way a user would receive this warning even if the quantum circuit was fine, right? Could an error be raised if a user creates a mutable QNode? (E.g. hashing the circuit operations: name of the operation and wire they act on and raising an error if the latest hash differs from the stored hash for the circuit.)